examples/stress_tests/particles.rs

```diff
@@ -0,0 +1,226 @@
+    let session = session_builder.start_p2p_session(socket)?;
+
+    App::new()
+        .add_plugins(GgrsPlugin::<Config>::default())
+        .set_rollback_schedule_fps(args.fps)
+        .add_systems(ReadInputs, read_local_inputs)
+        // SpriteBundle types
+        .rollback_component_with_clone::<Sprite>()
+        .rollback_component_with_clone::<Transform>()
+        .rollback_component_with_clone::<GlobalTransform>()
+        .rollback_component_with_clone::<Handle<Image>>()
+        .rollback_component_with_clone::<Visibility>()
+        .rollback_component_with_clone::<ComputedVisibility>()
+        // Also add our own types
+        .rollback_component_with_copy::<Velocity>()
+        .rollback_component_with_copy::<Ttl>()
+        .rollback_resource_with_clone::<ParticleRng>()
+        .checksum_component_with_hash::<Velocity>()
+        // todo: ideally we'd also be doing checksums for Transforms, but that's
+        // currently very clunky to do.
+        .insert_resource(args)
+        .add_plugins(DefaultPlugins.set(WindowPlugin {
+            primary_window: Some(Window {
+                resolution: WindowResolution::new(720., 720.),
+                title: "GGRS particles stress test".to_owned(),
+                ..default()
+            }),
+            ..default()
+        }))
+        .add_systems(Startup, setup)
+        .add_systems(Startup, spawn_particles) // spawn an initial burst of particles
+        .add_systems(
+            GgrsSchedule,
+            (
+                spawn_particles.run_if(spawn_pressed),
+                update_particles,
+                despawn_particles,
+            ),
+        )
+        .insert_resource(Session::P2P(session))
+        .insert_resource(ClearColor(Color::BLACK))
+        .insert_resource(ParticleRng(GameRng::seed_from_u64(123)))
+        .add_systems(Update, print_events_system)
+        .run();
+
+    Ok(())
+}
+
+fn setup(mut commands: Commands) {
+    commands.spawn(Camera2dBundle::default());
+}
+
+fn spawn_pressed(inputs: Res<PlayerInputs<Config>>) -> bool {
+    inputs.iter().any(|(i, _)| *i & INPUT_SPAWN != 0)
+}
+
+fn spawn_particles(mut commands: Commands, args: Res<Args>, mut rng: ResMut<ParticleRng>) {
+    let s = 200.0;
+    let ttl = args.fps * 5;
+
+    for _ in 0..args.rate {
+        commands
+            .spawn((
+                SpriteBundle {
+                    sprite: Sprite {
+                        color: Color::ORANGE,
+                        custom_size: Some(Vec2::splat(5.0)),
+                        ..default()
+                    },
+                    ..default()
+                },
+                Velocity(vec3(rng.gen_range(-s..s), rng.gen_range(-s..s), 0.0)),
+                Ttl(ttl),
+            ))
+            .add_rollback();
+    }
+}
+
+fn update_particles(mut particles: Query<(&mut Transform, &mut Velocity)>, args: Res<Args>) {
+    let time_step = 1.0 / args.fps as f32; // todo: replace with bevy_ggrs resource?
```

Comment:
On this theme, I think it would be good to replace the [Time](https://docs.rs/bevy/latest/bevy/time/struct.Time.html) resource during advance-frame with one that `bevy_ggrs` manually controls the time of. Then, systems already using the canonical Bevy `Time` resource would just automatically be rollback aware.